common/achievement/create.ts

```diff
@@ -0,0 +1,78 @@
+import { Prisma } from '@prisma/client';
+import { Achievement_template, JsonFilter } from '../../graphql/generated';
+import { ActionID, SpecificNotificationContext } from '../notification/actions';
+import { prisma } from '../prisma';
+import { TemplateSelectEnum, getAchievementTemplates } from './template';
+
+async function doesUserAchievementAlreadyExist<ID extends ActionID>(templateId: number, userId: string, context: SpecificNotificationContext<ID>) {
```

Comment:
Please update naming. `doesXXX` implies that it returns a boolean. You could use something like `findUserAchievementIfExists`.

Reply:
I think findUserAchievement would do the trick since a find function also could return undefined if nothing is found.